nebula_hw_interfaces/include/nebula_hw_interfaces/nebula_hw_interfaces_hesai/hesai_cmd_response.hpp

```diff
@@ -251,157 +253,189 @@ struct HesaiInventory
 };
 
 /// @brief struct of PTC_COMMAND_GET_CONFIG_INFO
-struct HesaiConfig
+struct HesaiConfigBase
 {
-  uint8_t ipaddr[4];
-  uint8_t mask[4];
-  uint8_t gateway[4];
-  uint8_t dest_ipaddr[4];
-  big_uint16_buf_t dest_LiDAR_udp_port;
-  big_uint16_buf_t dest_gps_udp_port;
-  big_uint16_buf_t spin_rate;
-  uint8_t sync;
-  big_uint16_buf_t sync_angle;
-  big_uint16_buf_t start_angle;
-  big_uint16_buf_t stop_angle;
-  uint8_t clock_source;
-  uint8_t udp_seq;
-  uint8_t trigger_method;
-  uint8_t return_mode;
-  uint8_t standby_mode;
-  uint8_t motor_status;
-  uint8_t vlan_flag;
-  big_uint16_buf_t vlan_id;
-  uint8_t clock_data_fmt;  // FIXME: labeled as gps_nmea_sentence in AT128, OT128 datasheets
-  uint8_t noise_filtering;
-  uint8_t reflectivity_mapping;
-  uint8_t reserved[6];
-
-  friend std::ostream & operator<<(std::ostream & os, nebula::HesaiConfig const & arg)
+  struct Internal{
+    uint8_t ipaddr[4];
+    uint8_t mask[4];
+    uint8_t gateway[4];
+    uint8_t dest_ipaddr[4];
+    big_uint16_buf_t dest_LiDAR_udp_port;
+    big_uint16_buf_t dest_gps_udp_port;
+    big_uint16_buf_t spin_rate;
+    uint8_t sync;
+    big_uint16_buf_t sync_angle;
+    big_uint16_buf_t start_angle;
+    big_uint16_buf_t stop_angle;
+    uint8_t clock_source;
+    uint8_t udp_seq;
+    uint8_t trigger_method;
+    uint8_t return_mode;
+    uint8_t standby_mode;
+    uint8_t motor_status;
+    uint8_t vlan_flag;
+    big_uint16_buf_t vlan_id;
+  };
+
+  json to_json()
   {
-    std::ios initial_format(nullptr);
-    initial_format.copyfmt(os);
+    json j;
+    std::stringstream ss;
+    ss << static_cast<int>(get().ipaddr[0]) << "."
+       << static_cast<int>(get().ipaddr[1]) << "."
+       << static_cast<int>(get().ipaddr[2]) << "."
+       << static_cast<int>(get().ipaddr[3]);
+    j["ipaddr"] = ss.str();
+    ss << static_cast<int>(get().mask[0]) << "."
```

Comment:
Please use a new `std::stringstream` for each address.
If you want to reuse the same name `ss`, you can wrap it in a new scope like
```cpp
{
  std::stringstream ss;
  ss << ...;
  j[...] = ss.str();
}
{
  std::stringstream ss; // Is a new stringstream
  ss << ...;
  ...
}

```